scipy/stats/tests/test_hypotests.py

```diff
@@ -1238,6 +1238,25 @@ def test_row_or_col_zero(self, input_sample, expected):
         assert_equal(pvalue, expected[0])
         assert_equal(statistic, expected[1])
 
+    def test_two_sided_gt_1(self):
+        # Check that returned p-value does not exceed 1 even when twice
+        # the minimum of the one-sided p-values does. See gh-15345.
+        tbl = [[1, 1], [13, 12]]
+        pl = boschloo_exact(tbl, alternative='less').pvalue
+        pg = boschloo_exact(tbl, alternative='greater').pvalue
+        assert 2*min(pl, pg) > 1
+        pt = boschloo_exact(tbl, alternative='two-sided').pvalue
+        assert pt == 1.0
+
+    @pytest.mark.parametrize("alternative", ("less", "greater"))
+    def test_against_fisher_exact(self, alternative):
+        # Check that the statistic of `boschloo_exact` is the same as the
+        # p-value of `fisher_exact` (for one-sided tests). See gh-15345.
```

Comment:
This was questioned in gh-15345, and it seems like a reasonable test to have included anyway, so I added it.